libgdf/src/gdf_table.cuh

```diff
@@ -454,9 +478,38 @@ public:
           }
       }
     }
+    return GDF_SUCCESS;
   }
 
+  __device__
+  void* get_column_device_pointer(size_type column_index)
+  {
+    return d_columns_data[column_index];
+  }
 
+  __device__
+  gdf_error get_row_valids(size_type row_index, byte_type * row_valid_byte_buffer) const
+  {
+    if(nullptr == row_valid_byte_buffer) {
+      return GDF_DATASET_EMPTY;
+    }
+    
+    for(size_type i = 0; i < num_columns; i++)
+    {
+      byte_type const * const d_col_valid_ptr = reinterpret_cast<byte_type const*> (d_columns_valids[i]);
```

Comment:
This cast will be unnecessary if `get_bit` is replaced by `gdf_is_valid` and `set_bit` is updated to accept a `gdf_valid_type*`.